### PR TITLE
Emails

### PR DESCRIPTION
**Pull request by betrevisan**
The commits proposed in this PR implement sending out emails. Changes include sending an email to the user once they first register, sending an email to both tutor and student when a tutoring session is scheduled, and sending an email to both tutor and student when a tutoring session is cancelled. Java tests were updated to check if the email was successfully sent.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/DeleteTutorSessionServlet.java [hunk 1]
```diff
@@ -44,12 +55,92 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         String studentID = Optional.ofNullable((String)request.getSession(false).getAttribute("userId")).orElse("-1");
         long id = Long.parseLong(request.getParameter("id"));
 
+        TutorSession tutorSession = datastore.getScheduledSession(id);
+
+        System.out.println(tutorSession.getStudentID());
```
Comment (kashiish) [inline src/main/java/com/google/sps/servlets/DeleteTutorSessionServlet.java:60]: Delete?

File: src/main/java/com/google/sps/servlets/DeleteTutorSessionServlet.java [hunk 2]
```diff
@@ -44,12 +55,92 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
         datastore.deleteTutorSession(studentID, id);
 
         String json = new Gson().toJson(datastore.getScheduledSession(id));
         response.setContentType("application/json;");
-        response.getWriter().println(json);
+        response.getWriter().println(json + "testTutorEmail: " + testTutorEmail + " testStudentEmail: " + testStudentEmail);
         return;
     }
+
+    public boolean sendCancellationEmailToTutor(Entity studentEntity, Entity tutorEntity, TimeRange timeslot) {
```
Comment (kashiish) [inline src/main/java/com/google/sps/servlets/DeleteTutorSessionServlet.java:78]: If the only different between `sendCancellationEmailToTutor` and `sendCancellationEmailToStudent` is the message, maybe we can make `message` a parameter and use a single method. I think we only use the studentEntity in the tutor method for the student's name and the tutorEntity in the student method for the tutor's name. We could also do all of the date stuff outside of this method to create the message.

So we could have the method be `sendCancellationEmail(Entity entity, String message)`. Does that make sense?

File: src/main/java/com/google/sps/servlets/DeleteTutorSessionServlet.java [hunk 3]
```diff
@@ -44,12 +55,92 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+        Properties props = new Properties();
+        Session session = Session.getDefaultInstance(props, null);
```
Comment (kashiish) [inline src/main/java/com/google/sps/servlets/DeleteTutorSessionServlet.java:80]: What is a `Session` here?

File: src/main/java/com/google/sps/servlets/SchedulingServlet.java [hunk 4]
```diff
@@ -71,8 +82,83 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+        boolean testStudentEmail = sendConfirmationEmailToStudent(studentEntity, tutorEntity, timeslot);
+
+
         String json = new Gson().toJson(datastore.getScheduledSessionsForTutor(tutorID));
-        response.getWriter().println(json);
+        response.getWriter().println(json + "testTutorEmail: " + testTutorEmail + " testStudentEmail: " + testStudentEmail);
         return;
     }
+
+    public boolean sendConfirmationEmailToTutor(Entity studentEntity, Entity tutorEntity, TimeRange timeslot) {
```
Comment (kashiish) [inline src/main/java/com/google/sps/servlets/SchedulingServlet.java:97]: We could do a similar thing here by consolidating these two methods into one?

File: src/main/java/com/google/sps/servlets/DeleteTutorSessionServlet.java [hunk 5]
```diff
@@ -44,12 +55,92 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+        } catch (MessagingException e) {
+            System.out.println("Failed to send email.");
+            return false;
+        } catch (UnsupportedEncodingException e) {
+            System.out.println("Failed to encode email.");
+            return false;
+        }
+    }
+
+    public boolean sendCancellationEmailToStudent(Entity studentEntity, Entity tutorEntity, TimeRange timeslot) {
```
Comment (kashiish) [inline src/main/java/com/google/sps/servlets/DeleteTutorSessionServlet.java:112]: Do these methods need to be public?

File: src/main/java/com/google/sps/servlets/SchedulingServlet.java [hunk 6]
```diff
@@ -71,8 +81,41 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
 
+        sendConfirmationEmail("bernardo.trevisan@gmail.com");
+
         String json = new Gson().toJson(datastore.getScheduledSessionsForTutor(tutorID));
         response.getWriter().println(json);
         return;
     }
+
+    public void sendConfirmationEmail(String to) {
+        System.out.println("Create email");
```
Comment (samfalberg) [inline src/main/java/com/google/sps/servlets/SchedulingServlet.java]: No longer needed?

File: src/main/java/com/google/sps/servlets/DeleteTutorSessionServlet.java (hunk 2 above)
Comment (betrevisan) [reply src/main/java/com/google/sps/servlets/DeleteTutorSessionServlet.java:78]: Yes, this makes sense. I'll implement this for the cancellation and scheduling emails.

File: src/main/java/com/google/sps/servlets/DeleteTutorSessionServlet.java (hunk 3 above)
Comment (betrevisan) [reply src/main/java/com/google/sps/servlets/DeleteTutorSessionServlet.java:80]: It is basically the session accessed by the email host to create the message and send it out. Here's more info about that: https://stackoverflow.com/questions/22657918/what-is-the-use-of-javax-mail-session

File: src/main/java/com/google/sps/servlets/DeleteTutorSessionServlet.java (hunk 5 above)
Comment (betrevisan) [reply src/main/java/com/google/sps/servlets/DeleteTutorSessionServlet.java:112]: Not at all, I'll make them private. Thanks!

**Pull request merged**